distributed/cli/tests/test_cli_utils.py

```diff
@@ -17,6 +17,7 @@ def test_uri_from_host_port():
     assert f('', 456, 123) == 'tcp://:456'
     assert f('', None, 123) == 'tcp://:123'
     assert f('', None, 0) == 'tcp://'
+    assert f('', 0, 123) == 'tcp://:0'
```

Comment:
The originally reported problem was here

```
mrocklin@carbon:~/workspace/distributed$ dask-scheduler  --port 0
distributed.scheduler - INFO - -----------------------------------------------
distributed.scheduler - INFO -   Scheduler at:  tcp://192.168.1.115:8786
distributed.scheduler - INFO -        http at:              0.0.0.0:9786
distributed.scheduler - INFO -       bokeh at:              0.0.0.0:8788
distributed.bokeh.application - INFO - Web UI: http://127.0.0.1:8787/status/
distributed.scheduler - INFO - -----------------------------------------------
```

It looks like this is due to `uri_from_host_port`

```python
In [1]: from distributed.cli.utils import uri_from_host_port

In [2]: uri_from_host_port('', 0, 8786)
Out[2]: 'tcp://:8786'
```

Which is used in dask-scheduler